micrometer-core/src/main/java/io/micrometer/core/instrument/binder/httpcomponents/PoolingHttpClientConnectionManagerMetricsBinder.java

```diff
@@ -0,0 +1,222 @@
+    private static final String METER_TOTAL_CONNECTIONS = "httpcomponents.httpclient.pool.total.connections";
+    private static final String METER_TOTAL_PENDING_DESC = "The number of connection requests being blocked awaiting a free connection for all routes.";
+    private static final String METER_TOTAL_PENDING = "httpcomponents.httpclient.pool.total.pending";
+    private static final String METER_DEFAULT_MAX_PER_ROUTE_DESC = "The configured default maximum number of allowed persistent connections per route.";
+    private static final String METER_DEFAULT_MAX_PER_ROUTE = "httpcomponents.httpclient.pool.route.max.default";
+    private static final String METER_ROUTE_MAX_DESC = "The configured maximum number of allowed persistent connections per route.";
+    private static final String METER_ROUTE_MAX = "httpcomponents.httpclient.pool.route.max";
+    private static final String METER_ROUTE_CONNECTIONS_DESC = "The number of persistent and leased connections per route.";
+    private static final String METER_ROUTE_CONNECTIONS = "httpcomponents.httpclient.pool.route.connections";
+    private static final String METER_ROUTE_PENDING_DESC = "The number of connection requests being blocked awaiting a free connection for all routes.";
+    private static final String METER_ROUTE_PENDING = "httpcomponents.httpclient.pool.route.pending";
+    private static final String TAG_CONNECTIONS_STATE = "state";
+
+    private final PoolingHttpClientConnectionManager connectionManager;
+    private final Iterable<Tag> tags;
+
+    private MultiGauge poolRouteMaxGauge;
+    private MultiGauge poolRouteAvailableGauge;
+    private MultiGauge poolRouteLeasedGauge;
+    private MultiGauge poolRoutePendingGauge;
+
+    /**
+     * Creates a metrics binder for the given pooling connection manager.
+     *
+     * For convenience this constructor will take care of casting the given
+     * {@link HttpClientConnectionManager} to the required {@link
+     * PoolingHttpClientConnectionManager}. An {@link IllegalArgumentException}
+     * is thrown, if the given {@code connectionManager} is not an instance of
+     * {@link PoolingHttpClientConnectionManager}.
+     *
+     * @param connectionManager The connection manager to monitor.
+     * @param name Name of the connection manager. Will be added as tag with the
+     *             key "httpclient".
+     * @param tags Tags to apply to all recorded metrics. Must be an even number
+     *             of arguments representing key/value pairs of tags.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public PoolingHttpClientConnectionManagerMetricsBinder(HttpClientConnectionManager connectionManager, String name, String... tags) {
+        this(connectionManager, name, Tags.of(tags));
+    }
+
+    /**
+     * Creates a metrics binder for the given pooling connection manager.
+     *
+     * For convenience this constructor will take care of casting the given
```

Comment:
I think it would be better to ask users to cast, so it is more clear what is required here without reading the JavaDoc - that is, enforced by the compiler. What is the scenario in which you're expecting to save users from casting themselves?